include/core/truthTable/truth_table.hpp

```diff
@@ -280,6 +329,18 @@ namespace syrec {
             cubeMap.insert(std::move(nh));
         }
 
+        static auto equal(TruthTable& tt1, TruthTable& tt2, const bool equalityUpToDontCare = true) -> bool {
+            if (!equalityUpToDontCare) {
+                return (tt1 == tt2);
+            }
+            for (auto const& [input, output]: tt1) {
+                if (!Cube::checkCubeEquality(tt1[input], tt2[input])) {
```

Comment:
With the latest changes this will raise an exception (or will simply not work) if `tt2` does not contain the input...
Please properly fix this so that comparison with missing inputs works in a symmetric fashion. Wouldn't hurt to add some proper tests that show that this actually works...

Reply:
Hope the changes in the following commit addresses the issue

Reply:
Yeah, thanks. Next time, try to include appropriate tests for such functionality.